src/pymmcore_widgets/_shutter_widget.py

```diff
@@ -0,0 +1,327 @@
+        if int(self._mmc.getProperty(self.shutter_device, "State")) == 0:
+            self.shutter_button.setIcon(icon(icon_c, color=self._icon_color_closed))
+        self._icon_closed = icon_c
+
+    @property
+    def icon_color_open(self) -> COLOR_TYPE:
+        """Set the button icon color for when the shutter is open."""
+        return self._icon_color_open
+
+    @icon_color_open.setter
+    def icon_color_open(self, color: COLOR_TYPE) -> None:
+        if int(self._mmc.getProperty(self.shutter_device, "State")) == 1:
+            self.shutter_button.setIcon(icon(self._icon_open, color=color))
+        self._icon_color_open = color
+
+    @property
+    def icon_color_closed(self) -> COLOR_TYPE:
+        """Set the button icon color for when the shutter is closed."""
+        return self._icon_color_closed
+
+    @icon_color_closed.setter
+    def icon_color_closed(self, color: COLOR_TYPE) -> None:
+        if int(self._mmc.getProperty(self.shutter_device, "State")) == 0:
+            self.shutter_button.setIcon(icon(self._icon_closed, color=color))
+        self._icon_color_closed = color
+
+    @property
+    def icon_size(self) -> int:
+        """Set the button icon size."""
+        return self._icon_size
+
+    @icon_size.setter
+    def icon_size(self, size: int) -> None:
+        self.shutter_button.setIconSize(QSize(size, size))
+        self._icon_size = size
+
+    @property
+    def button_text_open(self) -> str:
+        """Set the button text for when the shutter is open."""
+        return self._button_text_open
+
+    @button_text_open.setter
+    def button_text_open(self, text: str) -> None:
+        if int(self._mmc.getProperty(self.shutter_device, "State")) == 1:
+            self.shutter_button.setText(text)
+        self._button_text_open = text
+
+    @property
+    def button_text_closed(self) -> str:
+        """Set the button text for when the shutter is closed."""
+        return self._button_text_closed
+
+    @button_text_closed.setter
+    def button_text_closed(self, text: str) -> None:
+        if int(self._mmc.getProperty(self.shutter_device, "State")) == 0:
+            self.shutter_button.setText(text)
+        self._button_text_closed = text
+
+    def _create_wdg(self) -> None:
+
+        main_layout = QtW.QHBoxLayout()
+        main_layout.setContentsMargins(0, 0, 0, 0)
+        main_layout.setSpacing(3)
+
+        self.shutter_button = QtW.QPushButton(text=self._button_text_closed)
+        sizepolicy_btn = QtW.QSizePolicy(QtW.QSizePolicy.Fixed, QtW.QSizePolicy.Fixed)
+        self.shutter_button.setSizePolicy(sizepolicy_btn)
+        self.shutter_button.setIcon(
+            icon(self._icon_closed, color=self._icon_color_closed)
+        )
+        self.shutter_button.setIconSize(QSize(self._icon_size, self._icon_size))
+        self.shutter_button.clicked.connect(self._on_shutter_btn_clicked)
+        main_layout.addWidget(self.shutter_button)
+
+        if self.autoshutter:
```

Comment:
rather than actually changing the attributes this widget has, lets just do `if not self.autoshutter: self.autoshutter_checkbox.hide()`